infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/utils/LogReplicationConfigManager.java

```diff
@@ -288,54 +311,151 @@ private void generateFullTableConfig(LogReplicationSession session) {
             }
         });
 
-        LogReplicationFullTableConfig fullTableConfig = new LogReplicationFullTableConfig(session, streamsToReplicate,
-                streamToTagsMap, serverContext, streamsToDrop);
-        sessionToConfigMap.put(session, fullTableConfig);
-
-        log.info("LogReplicationFullTableConfig generated for session={}, streams to replicate={}, streamsToDrop={}",
-                TextFormat.shortDebugString(session), streamsToReplicate, streamsToDrop);
+        if (sessionToConfigMap.containsKey(session)) {
+            LogReplicationFullTableConfig config = (LogReplicationFullTableConfig) sessionToConfigMap.get(session);
+            config.setStreamsToReplicate(streamsToReplicate);
+            config.setDataStreamToTagsMap(streamToTagsMap);
+            config.setStreamsToDrop(streamsToDrop);
+            log.info("LogReplicationFullTableConfig updated for session={}, streams to replicate={}, streamsToDrop={}",
+                    TextFormat.shortDebugString(session), streamsToReplicate, streamsToDrop);
+        } else {
+            LogReplicationFullTableConfig fullTableConfig = new LogReplicationFullTableConfig(session, streamsToReplicate,
+                    streamToTagsMap, serverContext, streamsToDrop);
+            sessionToConfigMap.put(session, fullTableConfig);
+            log.info("LogReplicationFullTableConfig generated for session={}, streams to replicate={}, streamsToDrop={}",
+                    TextFormat.shortDebugString(session), streamsToReplicate, streamsToDrop);
+        }
     }
 
     /**
-     * If registry table has updates, update the streams to replicate and tags for each session based
-     * on the latest registry table entries.
+     * Get updated LogReplicationConfig for the given session.
+     *
+     * @param session LogReplicationSession to get updated config for.
+     * @param registryTableOnly True if only registry table is used for config generation or update.
      */
-    public void getUpdatedConfig() {
-        // Check if the registry table has new entries.  Otherwise, no update is necessary.
-        if (syncWithRegistryTable()) {
-            generateConfig(sessionToConfigMap.keySet());
+    public void getUpdatedConfig(LogReplicationSession session, boolean registryTableOnly) {
```

Comment:
It seems that a separate flag for 'registryTableOnly' is not required and the information can be derived from the model itself